[ClientAPI]ReaderBuilder.loadConf() not working

**Describe the bug**
Cannot create new reader while using loadConf method in reader builder chain.

**To Reproduce**
Steps to reproduce the behavior:
1. Create PulsarClient
2. `client.newReader().topic(tp).startMessageId(start).loadConf(readerConf).create()`

**Expected behavior**
Create a reader with both settings specified using methods as well as `loadConf`.

**Screenshots**
```
java.lang.RuntimeException: Failed to load config into existing configuration data
.....
Caused by: org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.apache.pulsar.client.api.MessageId` (no Creators, like default construct, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
 at [Source: (String)"{"cryptoFailureAction":"FAIL","topicName":"topic-0","readCompacted":false,"startMessageId":{"ledgerId":-1,"entryId":-1,"partitionIndex":-1},"receiverQueueSize":1000}"; line: 1, column: 92] (through reference chain: org.apache.pulsar.client.impl.conf.ReaderConfigurationData["startMessageId"])
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67) ~[pulsar-client-all-2.4.0-c509f86ba.jar:2.4.0-c509f86ba]
```

**Desktop (please complete the following information):**
 - OS: MacOS

**Additional context**
This may caused by `ReaderConfigurationData` contains a `startMessageId` of `MessageId` field that cannot be constructed. 
As `abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information` implies.

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ReaderBuilderImpl.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ReaderBuilderImpl.java
@@ -96,7 +96,9 @@ public class ReaderBuilderImpl<T> implements ReaderBuilder<T> {
 
     @Override
     public ReaderBuilder<T> loadConf(Map<String, Object> config) {
+        MessageId startMessageId = conf.getStartMessageId();
         conf = ConfigurationDataUtils.loadData(config, conf, ReaderConfigurationData.class);
+        conf.setStartMessageId(startMessageId);
         return this;
     }
 

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ReaderConfigurationData.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ReaderConfigurationData.java
@@ -18,6 +18,7 @@
  */
 package org.apache.pulsar.client.impl.conf;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import java.io.Serializable;
 
 import org.apache.pulsar.client.api.ConsumerCryptoFailureAction;
@@ -31,6 +32,8 @@ import lombok.Data;
 public class ReaderConfigurationData<T> implements Serializable, Cloneable {
 
     private String topicName;
+
+    @JsonIgnore
     private MessageId startMessageId;
 
     private int receiverQueueSize = 1000;

--- a/pulsar-client/src/test/java/org/apache/pulsar/client/impl/BuildersTest.java
+++ b/pulsar-client/src/test/java/org/apache/pulsar/client/impl/BuildersTest.java
@@ -21,7 +21,12 @@ package org.apache.pulsar.client.impl;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertTrue;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.pulsar.client.api.MessageId;
 import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.impl.conf.ReaderConfigurationData;
 import org.testng.annotations.Test;
 
 public class BuildersTest {
@@ -68,5 +73,28 @@ public class BuildersTest {
         builder = (ClientBuilderImpl)PulsarClient.builder().serviceUrl("pulsar+ssl://service:6650").enableTls(false);
         assertEquals(builder.conf.isUseTls(), false);
         assertEquals(builder.conf.getServiceUrl(), "pulsar+ssl://service:6650");
+    }
+
+    @Test
+    public void readerBuilderLoadConfTest() throws Exception {
+        PulsarClient client = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
+        String topicName = "test_src";
+        MessageId messageId = new MessageIdImpl(1, 2, 3);
+        Map<String, Object> config = new HashMap<>();
+        config.put("topicName", topicName);
+        config.put("receiverQueueSize", 2000);
+        ReaderBuilderImpl<byte[]> builder = (ReaderBuilderImpl<byte[]>) client.newReader()
+            .startMessageId(messageId)
+            .loadConf(config);
+
+        Class<?> clazz = builder.getClass();
+        Field conf = clazz.getDeclaredField("conf");
+        conf.setAccessible(true);
+        Object obj = conf.get(builder);
+        assertTrue(obj instanceof ReaderConfigurationData);
+        if (obj instanceof ReaderConfigurationData) {
+            assertEquals(((ReaderConfigurationData) obj).getTopicName(), topicName);
+            assertEquals(((ReaderConfigurationData) obj).getStartMessageId(), messageId);
+        }
     }
 }

